.circleci/config.yml

```diff
@@ -45,6 +45,31 @@ jobs:
       - restore_cache:
           key: vendor-{{ checksum "Gopkg.lock" }}
       - run: go build -o machine-controller github.com/kubermatic/machine-controller/cmd/controller
+      - save_cache:
+          key: machine-controller-{{ .Revision }}
+          paths:
+            - machine-controller
+
+  publish:
+    machine: true
```

Comment:
why machine instead of
```
    docker:
      - image: docker:stable
```
?

Reply:
Because that's what I found in the docs: https://circleci.com/docs/2.0/building-docker-images/#example